cartography/intel/aws/ec2/instances.py

```diff
@@ -21,410 +29,343 @@
+                {
+                    'InstanceId': instance_id,
+                    'ReservationId': reservation_id,
+                    'PublicDnsName': instance.get("PublicDnsName"),
+                    'PublicIpAddress': instance.get("PublicIpAddress"),
+                    'PrivateIpAddress': instance.get("PrivateIpAddress"),
+                    'ImageId': instance.get("ImageId"),
+                    'InstanceType': instance.get("InstanceType"),
+                    'IamInstanceProfile': instance.get("IamInstanceProfile", {}).get("Arn"),
+                    'MonitoringState': instance.get("Monitoring", {}).get("State"),
+                    'LaunchTime': instance.get("LaunchTime"),
+                    'LaunchTimeUnix': launch_time_unix,
+                    'State': instance.get("State", {}).get("Name"),
+                    'AvailabilityZone': instance.get("Placement", {}).get("AvailabilityZone"),
+                    'Tenancy': instance.get("Placement", {}).get("Tenancy"),
+                    'HostResourceGroupArn': instance.get("Placement", {}).get("HostResourceGroupArn"),
+                    'Platform': instance.get("Platform"),
+                    'Architecture': instance.get("Architecture"),
+                    'EbsOptimized': instance.get("EbsOptimized"),
+                    'BootMode': instance.get("BootMode"),
+                    'InstanceLifecycle': instance.get("InstanceLifecycle"),
+                    'HibernationOptions': instance.get("HibernationOptions", {}).get("Configured"),
+                },
+            )
 
+            subnet_id = instance.get('SubnetId')
+            if subnet_id:
+                subnet_list.append(
+                    {
+                        'SubnetId': subnet_id,
+                        'InstanceId': instance_id,
+                    },
+                )
 
-@timeit
-def load_ec2_instance_network_interfaces(neo4j_session: neo4j.Session, instance_data: Dict, update_tag: int) -> None:
-    neo4j_session.write_transaction(_load_ec2_instance_net_if_tx, instance_data, update_tag)
+            if instance.get("KeyName"):
+                key_name = instance["KeyName"]
+                key_pair_arn = f'arn:aws:ec2:{region}:{current_aws_account_id}:key-pair/{key_name}'
+                keypair_list.append({
+                    'KeyPairArn': key_pair_arn,
+                    'KeyName': key_name,
+                    'InstanceId': instance_id,
+                })
+
+            if instance.get("SecurityGroups"):
+                for group in instance["SecurityGroups"]:
+                    sg_list.append(
+                        {
+                            'GroupId': group['GroupId'],
+                            'InstanceId': instance_id,
+                        },
+                    )
 
+            for network_interface in instance['NetworkInterfaces']:
+                for security_group in network_interface['Groups']:
+                    network_interface_list.append({
```

Comment:
@resilience-jychp - remember how we talked about cardinality? This is how I'm dealing with it in a transform function. In this case, an instance can have multiple network interfaces, and each of those can have multiple security groups, so I'll just add them each as their own entry to the list.

Reply:
@achantavy did the same, not totaly convinced it's the best way to do it, but definitively do the job